File: Aula_0622/Lista07Ex02.py
class Frete:
  def __init__(self):
    self.__distancia = 0
    self.__peso = 0
  @property                           # get
  def Distancia(self):
    return self.__distancia
  @Distancia.setter                   # set
  def Distancia(self, distancia):
    if distancia > 0:
      self.__distancia = distancia
  @property                           # get
  def Peso(self):
    return self.__peso
  @Peso.setter                        # set
  def Peso(self, peso):
    if peso > 0:
      self.__peso = peso
  @property                           # get
  def ValorFrete(self):
    return 0.01 * self.__peso * self.__distancia
  
  def __str__(self):
    return f"Distancia = {self.__distancia}, Peso = {self.__peso}"

if __name__ == "__main__":
  f = Frete()
  f.Distancia = 10
  f.Peso = 50
  valor = f.ValorFrete
  print(f)
  print(f"Valor do frete = R$ {valor:0.2f}")

File: Aula_0622/Lista07Ex02.cs
using System;

class Program {
  public static void Main() {
    Frete f = new Frete();
    f.Distancia = 10;
    f.Peso = 50;
    double valor = f.ValorFrete;
    Console.WriteLine(f);
    Console.WriteLine($"Valor do frete = R$ {valor:0.00}");
  }
}
class Frete {
  private double distancia, peso;
  public double Distancia {
    set { if (value > 0) distancia = value; }  
    get { return distancia; }
  }
  public double Peso {
    set => peso = value > 0 ? value : 0;
    get => peso;
  }
  public double ValorFrete {
     get { return 0.01 * distancia * peso; }   
  }
  public override string ToString() {
    return $"Distancia = {distancia}, Peso = {peso}";
  }
}